static/js/stream_edit.js

```diff
@@ -585,6 +585,40 @@ $(function () {
         e.stopPropagation();
     });
 
+    $("#subscriptions_table").on("click", ".deactivate", function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        var stream_id = get_stream_id(e.target);
+        if (!stream_id) {
+            ui_report.message(i18n.t("Invalid stream id"), $(".stream_change_property_info"), 'alert-error');
+            return;
+        }
+        var stream_name = stream_data.maybe_get_stream_name(stream_id);
+        var deactivate_stream_modal = templates.render("deactivation-stream-modal");
+        $(".subscription_settings").append(deactivate_stream_modal);
+        $("#deactivation_stream_modal .stream_name").text(stream_name);
+        overlays.open_modal('deactivation_stream_modal');
+    });
+
+    $("#subscriptions_table").on("click", "#do_deactivate_stream_button", function (e) {
+        var stream_id = get_stream_id(e.target);
+        if (!stream_id) {
+            ui_report.message(i18n.t("Invalid stream id"), $(".stream_change_property_info"), 'alert-error');
+            return;
+        }
+        overlays.close_modal('deactivation_stream_modal');
+        var row = $(".stream-row.active");
+        settings_streams.delete_stream(stream_id, $(".stream_change_property_info"), row);
+        $(".settings").hide();
+        $(".nothing-selected").show();
+        $("#deactivation_stream_modal").remove();
+    });
+
+    $("#subscriptions_table").on("hide.bs.modal", "#deactivation_stream_modal", function () {
+        $("#deactivation_stream_modal").remove();
+    });
+
```

Comment:
The reason of setting event on `hide.bs.modal` rather than `click event on data-dismiss` is that, data-dismiss event is only get called when modal is closed from click on `Cancel` btn, not when user press `Esc` and close the modal. We should remove modal element from DOM in that case also. `hide.bs.modal` will get called every time when modal is closed.

Reply:
OK, makes sense.